app/data-store/reducers/calendar-reducer.js

```diff
@@ -16,14 +16,16 @@ export default (state = {}, action) => {
 
         case 'NAVIGATE':
             if (state.router.route == 'index') {
-                // page is now ready, initialize the calendar...
-                setTimeout(() => {
-                    renderCalendar();
-                },100);
-
+                setTimeout(renderCalendar,1,state.sources);
+                setTimeout(renderEventsOnNavigate,1,state.sources);
             }
             return state;
 
+        case 'TOGGLEROOM':
+
+            Object.assign({}, state, state.sources[action.value.room - 1].visible = !action.value.visible );
```

Comment:
The Object assign() called above isn't doing anything (there is important work there but not with the assign, it's only the assignment that is mutating the array which makes any change. Because Object.assign() is returning a new object but it's not being assigned to any variable anyways.

`state.sources[action.value.room -1].visible = !action.value.visible`

A way to do this without mutating the state might be like the following... however this fixes the state and the state shows up fine, but the calendars then don't toggle. See if you can spot the why and where, if I have time later too I'll try to help out.
```js
      case 'NAVIGATE':
          // Get the value for "visible", "room" and "sources"
          const {value: {visible, room}} = action;
          const {sources} = state;
          
          // Get the index of room in "sources" array
          const idx = room-1;
          // create a new room object with changed visible property
          const newRoom = Object.assign({}, sources[idx], {visible: !visible});
          // Create a new sources Array with all old rooms and the newRoom.
          const newSources = [].concat(sources.slice(0, idx), newRoom, sources.slice(idx+1));
          
          // Return the state object with no changes
          return Object.assign({}, state, {sources: newSources});
```